### PR TITLE
Fixes #17567: First inventory fails when installing root server

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/17567

### CODE REVIEW

File: share/commands/agent-reset [hunk 1]
```diff
@@ -81,18 +78,18 @@ if [ -f "${RUDDER_VAR}/cfengine-community/policy_server.dat" ]
...
     rudder agent update "$@"
   else
-    # Ensure that ncf is up to date  
-    [ "$VERBOSE" = true ] && echo "Ensure ncf is up to date..."
-    # $@ to pass options through
-    rudder agent update "$@"
+    # Initial promises
+    [ "$VERBOSE" = true ] && echo "Restoring initial promises..."
+    cp -r /opt/rudder/share/initial-promises/* /var/rudder/cfengine-community/inputs/
+    cp -r /usr/share/ncf/tree/* /var/rudder/ncf/common/
```
Comment (ncharles) [inline share/commands/agent-reset:92]: why not rudder agent update instead of copying the folder with cp ?

Comment (ncharles) [reply share/commands/agent-reset:92]: especially as you are removing /var/rudder/ncf/local

File: share/commands/agent-reset [hunk 2]
```diff
@@ -54,14 +54,11 @@ done
...
 # - restore initial promises
-if [ "${UUID}" != "root" ]
-then
-  [ "$VERBOSE" = true ] && echo "Restoring initial promises..."
-  rm -rf /var/rudder/cfengine-community/inputs/*
-  # - remove ncf
-  rm -rf /var/rudder/ncf/common/*
-  rm -rf /var/rudder/ncf/local/*
-fi
+[ "$VERBOSE" = true ] && echo "Restoring initial promises..."
```
Comment (amousset) [inline share/commands/agent-reset:57]: policies

File: share/commands/agent-reset [hunk 3]
```diff
@@ -81,18 +78,18 @@ if [ -f "${RUDDER_VAR}/cfengine-community/policy_server.dat" ]
...
     [ "$VERBOSE" = true ] && echo "Getting initial promises..."
     # $@ to pass options through
     rudder agent update "$@"
   else
-    # Ensure that ncf is up to date  
-    [ "$VERBOSE" = true ] && echo "Ensure ncf is up to date..."
-    # $@ to pass options through
-    rudder agent update "$@"
+    # Initial promises
+    [ "$VERBOSE" = true ] && echo "Restoring initial promises..."
```
Comment (amousset) [inline share/commands/agent-reset:90]: policies

File: share/commands/agent-reset (hunk 1 above)
Comment (peckpeck) [reply share/commands/agent-reset:92]: because we need working promises to update

Comment (ncharles) [reply share/commands/agent-reset:92]: you already copied initial promises, so update works

**Comment by peckpeck** [comment]
PR updated with a new commit

File: share/commands/agent-reset (hunk 2 above)
Comment (amousset) [inline share/commands/agent-reset:57]: policies

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Comment by peckpeck** [comment]
OK, merging this PR

**Pull request merged**